src/sinks/util/auto_concurrency/controller.rs

```diff
@@ -131,37 +138,47 @@ impl<L> Controller<L> {
             in_flight: inner.in_flight as u64
         });
 
-        let current_rtt = inner.current_rtt.update(rtt);
+        if use_rtt {
+            inner.current_rtt.update(rtt);
+        }
+        let current_rtt = inner.current_rtt.average();
         let past_rtt = inner.past_rtt.average();
 
+        // No past measurements, set up initial values.
         if past_rtt == 0.0 {
```

Comment:
Have you considered making `EWMA::average` return `Option` like `Mean`? It looks like it would be better for consistency, and it'll also make for a better-structured use of the value - is would ensure there'd be no way to interpret the first measurement incorrectly, and address other problems of that sort.

Reply:
I hadn't, but that makes sense.